uw-frame-components/portal/messages/services.js

```diff
@@ -0,0 +1,240 @@
+'use strict';
+
+define(['angular'], function(angular) {
+  return angular.module('portal.messages.services', [])
+    .factory('messagesService', [
+      '$http',
+      '$log',
+      '$localStorage',
+      '$sessionStorage',
+      '$q',
+      '$filter',
+      'portalGroupService',
+      'miscService',
+      'keyValueService',
+      'SERVICE_LOC',
+      'MESSAGES',
+      'KV_KEYS',
+      function($http,
+               $log,
+               $localStorage,
+               $sessionStorage,
+               $q,
+               $filter,
+               portalGroupService,
+               miscService,
+               keyValueService,
+               SERVICE_LOC,
+               MESSAGES,
+               KV_KEYS) {
+        // //////////////////
+        // Exposed methods //
+        // //////////////////
+
+        /**
+         * @typedef {Object} Message
+         * @property {number} id
+         * @property {string} title
+         * @property {string} titleShort
+         * @property {string} description
+         * @property {string} descriptionShort
+         * @property {string} messageType
+         * @property {string} goLiveDate
+         * @property {string} expireDate
+         * @property {string} featureImageUrl
+         * @property {string} priority
+         * @property {Object} actionButton
+         * @property {Object} moreInfoButton
+         * @property {Object} confirmButton
+         */
+
+        /**
+         * Get everything at the messages service endpoint
+         * @returns {Array} An array of message objects
+         */
+        var getAllMessages = function() {
+          return $http.get(SERVICE_LOC.messagesURL)
+            .then(function(response) {
+              return response.data;
+            })
+            .catch(function(error) {
+              miscService.redirectUser(error.status, 'Get all messages');
+            });
+        };
+
+        /**
+         * Filter the array of messages based on each message's groups
+         * attribute
+         * @param messages An array of message objects
+         * @returns {Array} A filtered array of messages
+         */
+        var getMessagesByGroup = function(messages) {
+          return portalGroupService.getGroups()
+            .then(function(groups) {
+              var messagesByGroup = [];
+              angular.forEach(messages, function(message) {
+                var added = false;
+                // For each group for the current notification
+                angular.forEach(message.audienceFilter.groups,
```

Comment:
Probably need a special case where if the message has no groups or its groups array is empty, consider it to apply to everyone ("not filtered by groups") and so include it for this user.